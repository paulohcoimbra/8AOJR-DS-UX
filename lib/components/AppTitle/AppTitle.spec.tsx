import React from 'react';
import { render, screen } from '@testing-library/react';
import { describe, it, expect } from 'vitest';
import '@testing-library/jest-dom';
import { AppTitle } from '.';

describe('AppTitle', () => {
  it('renders the title text', () => {
    render(<AppTitle title="My App" subtitle='This is my app' />);
    const title = screen.getByText('My App');
    expect(title).toBeInTheDocument();
  });

  it('renders as an h1 by default', () => {
    render(<AppTitle title="Default Heading" subtitle='This is my app' />);
    const titleElement = screen.getByText('Default Heading');
    expect(titleElement.tagName).toBe('H1');
  });


  it('matches snapshot', () => {
    const { container } = render(<AppTitle title="Snapshot Test" subtitle='This is my app' />);
    expect(container).toMatchSnapshot();
  });
});