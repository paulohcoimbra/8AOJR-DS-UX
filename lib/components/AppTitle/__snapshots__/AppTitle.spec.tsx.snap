// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`AppTitle > matches snapshot 1`] = `
<div>
  <div
    class="sc-gIIpAQ dAScAa"
  >
    <h1
      class="sc-VHjGu cuSVnV"
    >
      Snapshot Test
    </h1>
    <h3
      class="sc-dQkurY kZfLFP"
    >
      This is my app
    </h3>
  </div>
</div>
`;
